### PR TITLE
Fix for issue #7027.

### PR DESCRIPTION
**Pull request by kaichogami**
Changed `_diff_wrt` to `False` in case of defined functions while `True` for undefined functions. I have a feeling this might break code somewhere. @asmeurer please review and let me know if you think some additional tests might be required.

### CODE REVIEW

File: sympy/core/function.py [hunk 1]
```diff
@@ -345,7 +345,7 @@ class Function(Application, Expr):
 
     @property
     def _diff_wrt(self):
-        """Allow derivatives wrt functions.
+        """Don't Allow derivatives wrt defined functions.
```
Comment (smichr) [inline sympy/core/function.py]: Allow -> allow

**Comment by kaichogami** [comment]
@smichr @asmeurer  Three lines { [882](https://github.com/kaichogami/sympy/blob/diff_wrt_function/sympy/core/function.py#L882) , [884](https://github.com/kaichogami/sympy/blob/diff_wrt_function/sympy/core/function.py#L884) , [908](https://github.com/kaichogami/sympy/blob/diff_wrt_function/sympy/core/function.py#L908) } fail as a result of doc test which involves `_diff_wrt` defined functions. Is it best to remove or change the doc string accordingly?

**Comment by smichr** [comment]
In reading the docstring by the test that failed (line 823) it appears that this is a desired feature that you have disabled. So the failing test is a sign that this should not be done (and that's perhaps the reason it wasn't done in the first place). My guess is that this should just be closed. 

In the future I would recommend that you ask a question about some behavior in the mailing list or, if you have a case that is failing, open an issue. Finally, open a PR addressing an issue. 

Perhaps you had an issue that motivated this PR. It would help in evaluating what to do next if that issue were presented in an Issue or at least the opening paragraph (OP) of this PR.

**Comment by kaichogami** [comment]
@smichr I apologize for failing to write the reason at the beginning. 
I created this PR as a follow up of the issue #7027 with the intention of contributing to Sympy. Although I have not faced any issue myself, I got curious following the discussion in #4915, discussing about the issue of differentiating wrt functions. Clearly the current behavior is desired(at-least in sympy) and it definitely needs to be discussed before any change is proposed. My apologies again for creating a PR without appropriate reason.

**Comment by asmeurer** [comment]
This is a desired change. That's the whole point of issue https://github.com/sympy/sympy/issues/7027. You should delete the failing tests, as they are no longer testing desired behavior. @kaichogami let me know if you want to finish this.

**Comment by kaichogami** [comment]
@asmeurer alright, I will finish this.

**Comment by kaichogami** [comment]
If its not something urgent, @shibasisp would like to try his hands on it.

**Comment by shibasisp** [comment]
Yes, I am new to open source and I would like to work on this issue and start contributing.

**Comment by asmeurer** [comment]
Go for it.

**Pull request closed without merging**